### PR TITLE
feat: upgrade dependencies, add new options, add better error handling, add test

### LINKED ISSUE #19: Can't set window size for DMG-Installer
I am using ember electron to create my app , but i am having problem with macOS installer. To build my app i am using `ember electron:make` command and it takes options from eletron-forge-config.js file.I was trying to set a background image for dmg-installer, and as my image has size 600x571 , i set same window size.
But somehow it resized my image and my windows became 600x571, but image 450x427.
Then I tried to resize window size to 450x427 , but now it not resizing at all.
Here is my code now for dmg installer:
`electronInstallerDMG: { title: 'DyeGraph', background: 'public/assets/images/logo.png', icon: '/Users/roman.blyztsiv/projects/dyegraph/public/assets/images/logo.png', window: { width: 450, height: 427 }, contents: [ { 'x': 94, 'y': 260, 'type': 'file', 'path': '/Users/roman.blyztsiv/projects/dyegraph/electron-out/DyeGraph-darwin-x64/DyeGraph.app' }, { 'x': 355, 'y': 260, 'type': 'link', 'path': '/Applications' } ] },`
I was trying to make everything like [here](https://github.com/tryghost/Ghost-Desktop) , but nothing is happening

### LINKED ISSUE #26: Update node-appdmg to sign .dmg
hi!

it will be nice that electron-forge has the possibility of sign a dmg,

and for this reason, this module should update to last node-appdmg!

can it update?

thanks!!

### PR DESCRIPTION
**Pull request by MarshallOfSound**
One important thing this adds is a new catch all `additionalDMGOptions` option which allows people to use features of `node-appdmg` without us explicitly adding a new property and proxying it in this module.

Fixes #26
Fixes #19

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
The diff looks big but that's just because I ES6'ified a thing and then moved it to a new folder

**Comment by malept** [comment]
So `additionalDMGOptions` overwrites any other option?

**Comment by MarshallOfSound** [comment]
Yes, but that's just a foot gun if you do that :)

**Comment by malept** [comment]
Yeah, well I can just predict that we'll see issues about how some param doesn't work because the user overwrote it in the additional options. I deliberately put extra options first before named params in various parts in Packager.

**Comment by MarshallOfSound** [comment]
Updated 👍

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
 node_js:
-  - '0.12'
-  - 'iojs'
   - 6
-  - 7
-script: 'npm run-script ci'
+  - 8
+  - 10
+os: osx
+script: 'yarn lint && yarn test'
```
Comment (malept) [inline .travis.yml]: yarn? Perhaps `cache: yarn` then?

File: test/index.test.js [hunk 2]
```diff
@@ -1,8 +1,52 @@
-var createDMG = require('../');
-var assert = require('assert');
+const assert = require('assert');
+const electronDownload = require('electron-download');
```
Comment (malept) [inline test/index.test.js]: One option is to commit packaged app fixtures, like `electron-installer-debian`.

File: test/index.test.js [hunk 3]
```diff
@@ -1,8 +1,52 @@
...
+const fs = require('fs');
+const path = require('path');
+const rimraf = require('rimraf');
+const zip = require('cross-zip');
```
Comment (malept) [inline test/index.test.js]: Out of curiosity, why `cross-zip` and not `extract-zip`?

**Pull request merged**